include/aos/common/downloader.hpp

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_DOWNLOADER_HPP_
+#define AOS_DOWNLOADER_HPP_
+
+#include "aos/common/error.hpp"
+#include "aos/common/string.hpp"
+
+namespace aos {
+namespace downloader {
+
+class DownloaderItf {
+public:
+    /**
+     * Downloads file.
+     *
+     * @param url URL.
+     * @param path path to file.
+     * @param contentType content type.
+     * @return Error.
+     */
+    virtual Error Download(const aos::String& url, const aos::String& path, const aos::String& contentType) = 0;
```

Comment:
I forget, do we need content type? Could we use aos::EnumStringer for it?

Reply:
Done